### DISCUSSION TITLE
Bare minimum to insert/update container using VB.NET

### DISCUSSION THREAD

**Discussion by scotty562** [fuzzzerd/fmdata#166] [Q&A]
Please forgive my ignorance on this subject. I'm trying to insert/update PDF files into FileMaker containers using an existing VB.NET application. Is there a VB.NET example? I only see C# examples.

**Comment by fuzzzerd** [comment]
Are you intending to create a new record or update an existing record? That makes a difference. 

If you're creating a new record, you must first call the [`CreateAsync`](https://www.fuget.org/packages/FMData/4.3.0/lib/netstandard2.0/FMData.dll/FMData/FileMakerApiClientBase#M%3AFMData.FileMakerApiClientBase.CreateAsync%60%601%28%60%600%29) method first and use the returned RecordId to call [`UpdateContainerAsync`](https://www.fuget.org/packages/FMData/4.3.0/lib/netstandard2.0/FMData.dll/FMData/FileMakerApiClientBase#M%3AFMData.FileMakerApiClientBase.UpdateContainerAsync%28System.String%2CSystem.Int32%2CSystem.String%2CSystem.String%2CSystem.Byte%5B%5D%29) with a layout and a `byte[]` that to be pushed into the container.

Alternatively if the record exists, you can call `UpdateContainerAsync` by passing the RecordId of the existing record.

I don't have any VB examples, but when you get something working if you share it here I will merge in a PR that updates the readme doc.

**Reply by scotty562** [reply]
I will always be updating an existing record. Thank you for the info I'll see how far I can get.
